### PR TITLE
Docs: Revise location of some LogQL examples; improve prose

### PR DESCRIPTION
**Pull request by KMiller-Grafana**
This docs-only revision relocates some existing examples.  It also updates prose about the examples and about evaluation precedence for logical operators in the label filtering explanation.

Future PRs will better address the organization of the examples. This PR aims only to gather examples in a single location.

### CODE REVIEW

File: docs/sources/logql/query_examples.md [hunk 1]
```diff
@@ -62,3 +64,136 @@ and only include errors whose duration is above ten seconds.
...
+order the filtering stages left to right:
+
+1. stream selector
+2. line filters
+3. label filters
+
+Consider the query:
+
+```logql
+{cluster="ops-tools1", namespace="loki-dev", job="loki-dev/query-frontend"} |= "metrics.go" !="out of order" | logfmt | duration > 30s or status_code!="200"
```
Comment (MichelHollands) [inline docs/sources/logql/query_examples.md:81]:
```suggestion
{cluster="ops-tools1", namespace="loki-dev", job="loki-dev/query-frontend"} |= "metrics.go" != "out of order" | logfmt | duration > 30s or status_code != "200"
```

Comment (MichelHollands) [reply docs/sources/logql/query_examples.md:81]: (nit) The spaces between operator and operands are applied a bit randomly.

File: docs/sources/logql/metric_queries.md [hunk 2]
```diff
@@ -142,24 +120,4 @@ The aggregation operators can either be used to aggregate over all label values
...
-sum(count_over_time({job="mysql"}[5m])) by (level)
-```
-
-Get the rate of HTTP GET requests to the `/home` endpoint for NGINX logs by region:
-
-```logql
-avg(rate(({job="nginx"} |= "GET" | json | path="/home")[10s])) by (region)
-```
-
+See [vector aggregation examples](../query_examples/#vector-aggregation-examples) for query examples that use the vector aggregation.
```
Comment (ssncferreira) [inline docs/sources/logql/metric_queries.md:123]: Small nit 

```suggestion
See [vector aggregation examples](../query_examples/#vector-aggregation-examples) for query examples that use vector aggregation expressions.
```

File: docs/sources/logql/query_examples.md [hunk 3]
```diff
@@ -62,3 +64,136 @@ and only include errors whose duration is above ten seconds.
...
+```
+
+There are two line filters: 
+`|= "metrics.go"`
+and `!="out of order"`.
+Of the log lines identified with the stream selector,
+the query results
+include only those log lines that contain the string "metrics.go"
+and do not contain the string "out of order".
+
```
Comment (ssncferreira) [inline docs/sources/logql/query_examples.md:96]: It is missing the explanation of the `| logfmt` parser 👀 
In order to have access to labels `duration` and `status_code` this parser needs to be applied.

**Review by ssncferreira — COMMENTED** [review]
LGTM ✍️  Just some small comments

**Review by ssncferreira — APPROVED** [review]
LGTM, thanks 🙏

**Pull request merged**